testsuite/integration-arquillian/tests/other/base-ui/src/test/java/org/keycloak/testsuite/ui/account2/PersonalInfoTest.java

```diff
@@ -154,6 +155,11 @@ public void disabledEditUsername() {
         personalInfoPage.valuesEqual(testUser2);
     }
 
+    @Test
+    public void testNameInToolbar() {
+        assertEquals("test user", personalInfoPage.header().getToolbarLoggedInUser().getText());
```

Comment:
Please make sure the test is thorough enough and possibly covers all the possibilities (user doesn't have first and last name, user has only first or last name, user has preferred username, ...).

And possibly add a test for Welcome Screen as well.

Reply:
It always only the username, like it states in the issue

Reply:
> Currently, it's not easy to tell what user is logged in to the new Account Console. There's just "anonymous" avatar icon in the page header (which is ok for now as we don't support custom avatars) but no username.

Reply:
added more cases